source/distributed/solution_transfer.cc

```diff
@@ -424,7 +424,8 @@ namespace parallel
       for (; it_input != dof_values.cend(); ++it_input, ++it_output)
         cell->set_dof_values_by_interpolation(*it_input,
                                               *(*it_output),
-                                              fe_index);
+                                              fe_index,
+                                              true);
```

Comment:
You said in the changelog that there are conditions where you do not want to perform the check. Yet, you set it always to true.

You should give the user the decision to perform this check. Maybe set it to true by default, but give them the opportunity to disable it.

Reply:
You also said in the introduction that you encounter the problem in serial now as well. So would it make sense to also incorporate this change in the serial `SolutionTransfer` class?